### PR TITLE
Scenes tutorial

### PR DESCRIPTION
**Pull request by bglendenning**
To view this locally:

```
python -m vent .venv
source .venv/bin/activate
python -m pip install requests
python -m pip install jupyter
jupyter notebook
```

Query annotated scenes endpoint & then download GPKG of annotated scenes results.

[Formatted Scenes Notebook](https://github.com/carbon-mapper/platform-public/blob/fc6e052c782f8e9fb965640d7dc04078363b9f94/tutorials/scenes.ipynb)

### CODE REVIEW

File: tutorials/scenes.ipynb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    "date_range = '2020-01-01/2023-12-31'\n",
+    "response = requests.get(\n",
+    "    f\"https://api.carbonmapper.org/api/v1/catalog/scenes/annotated?intersects={geojson}&datetime={date_range}\",\n",
+    "    headers={\"Authorization\": f\"Bearer {token}\"},\n",
```
Comment (bglendenning) [inline tutorials/scenes.ipynb:64]: Need to figure out API token auth for the views that require it.

**Comment by alexdiamondbv** [comment]
Very nice, thanks! 

I was able to go through the tutorial successfully but only 100 of the 497 scenes were included in the Geopackage. Is this a limit we imposed somewhere?

Also, is it possible to remove "that are eligible for null detection," from this sentence:

> In this tutorial, we will use the annotated scenes endpoint to discover published scenes that are eligible for null detection, and then use other endpoints to download the scenes discovered.

**Comment by bglendenning** [comment]
I assume the 100 limit is related to pagination, so I can make a note about that. I can get the null detection line removed.

**Review by demiurg — APPROVED** [review]
I like this example and the use of libraries that I did not know existed. Looks good!

**Comment by demiurg** [comment]
I suppose this has to wait to get merged until we deploy download endpoint to production.

**Comment by demiurg** [comment]
Pagination is definitely a common "gotcha", and what people will run into. I think paginating through the results would be good to document as well, although it is awkward with a format like GPKG, not sure what is best for the user here.

**Comment by bglendenning** [comment]
Yes, and should also update this with the production API URL when it's available.

I think the 100 limit is coming from the annotated scenes endpoint. I think a short note about the pagination should be adequate. AFAIK the only limit for the download endpoint is the URL size.

**Comment by demiurg** [comment]
There is an implicit resource based limit, it just times out at some >1000 number of scenes.

**Comment by bglendenning** [comment]
Updated per Alex's review. New notebook can be viewed [here](https://github.com/carbon-mapper/platform-public/blob/fc6e052c782f8e9fb965640d7dc04078363b9f94/tutorials/scenes.ipynb).

**Comment by alexdiamondbv** [comment]
Thanks, the changes look good. I assume most developers can deal with the pagination. Although, it does seem a little odd to be exporting geopackages of only 100 scenes at a time. Most GIS and data science users would be analyzing at least 1000 at a time, if not more (e.g. all of CA, or all of the Permian Basin). Perhaps that's what the bulk downloads are for though.

**Comment by demiurg** [comment]
Perhaps it would be good to test in production, what kind of page limit do we hit while staying under 20 seconds for response.

**Comment by alexdiamondbv** [comment]
Would it make sense to add the same "naive pagination consumer" used in the plumes tutorial to the scenes tutorial?

**Comment by bglendenning** [comment]
> Thanks, the changes look good. I assume most developers can deal with the pagination. Although, it does seem a little odd to be exporting geopackages of only 100 scenes at a time. Most GIS and data science users would be analyzing at least 1000 at a time, if not more (e.g. all of CA, or all of the Permian Basin). Perhaps that's what the bulk downloads are for though.

> Would it make sense to add the same "naive pagination consumer" used in the plumes tutorial to the scenes tutorial?

What I've learned about geopackages so far, they're sqlite databases, so it's probably possible to extend the scope of the gpkg tutorial to include sqlite operations to create larger gpkg databases. I'm sure anybody from the CM team could extend the tutorial with that in mind.

**Pull request merged**